### ISSUE TITLE
Move core to a multi-platform Chatbot library

### ISSUE THREAD

**Issue by eknoes** [eknoes/covidbot#127]
After #117, it would be easy to extract the core of that bot and create a Python Library for Multi-Platform Chatbots.

**Comment by Lukas2112** [comment]
Is there anything new here?

**Comment by eknoes** [comment]
Unfortunately not, sorry. I am currently quite busy.

**Comment by Lukas2112** [comment]
Okay. Is there any guidance on how to combine the programmed into a library? Because maybe I could have a look at it.

**Comment by eknoes** [comment]
Not really, one has to make some architectural decisions. Currently, all the functionality is in __main__.py and the different platforms are in interfaces/. It would be nice to have a library with a common interface and decorators for the chatbot.

**Comment by Lukas2112** [comment]
https://medium.com/analytics-vidhya/how-to-create-a-python-library-7d5aea80cc3f

I found this post which seems to be quite helpful. What do you think, @eknoes ?

**Comment by eknoes** [comment]
Yes, the question is not really how it could be put into a library, but how the library API should be designed.

**Comment by Lukas2112** [comment]
> Yes, the question is not really how it could be put into a library, but how the library API should be designed.

Would much have to be changed structurally?

**Comment by eknoes** [comment]
Yes! Honestly, I am not sure if you can help a lot regarding this issue, but of course you can give it a try. One must decide how to e.g. annotate a bot who would use this library: Obviously, the functions would return BotResponses (https://github.com/eknoes/covidbot/blob/main/covidbot/interfaces/bot_response.py) and some kind of general structure for all messenger messages. Furthermore, it should provide hooks, so that prior to receiving a general state might be saved. Basically the setup and this function must be extracted (https://github.com/eknoes/covidbot/blob/2d4f1e43124090a3c5d23aa623554b813835f504/covidbot/bot.py#L111). The setup should be replaced through decorators.

**Comment by Lukas2112** [comment]
Have cleanups or similar changes to the code base changed anything in the initial situation, @eknoes ?

**Comment by eknoes** [comment]
No, there was no progress or heavy changes here. If you still want to help, you could come up with a suggestion for a general API and we can discuss it. I currently do even not have enough spare time to handle the core issues of the bot.
